### DISCUSSION TITLE
How to Read Input in ABS

### DISCUSSION THREAD

**Discussion by aulia-adil** [abstools/abstools#303] [Q&A] [answered]
Hello everyone, I want to know how to read input in ABS language like "Python input()". I have searched the manual with keyword "read" and "input" but  I still can't find a way to read input. So is there a way to read input in ABS language?

**Comment by rudi** [comment, marked as answer]
Hello, that's a good question.  As ABS was conceived as a modeling language, it originally did not contain input or output.  (Also, ABS was originally implemented on top of the Maude rewriting logic language, which at that time did not offer input or output facilities.)

Since then, and because the focus on ABS has shifted to simulation (as opposed to, e.g., model checking), we added a `print` expression.  Also, we added a way to communicate with the model via http requests, and I hope to add soon a way to read from an SQLite database.

But indeed, there is no way to prompt the user for some text and then parse it.  Usually, user behavior is added to the model instead.  I hope this was helpful to you; feel free to as follow-up questions if not.
